common/include/window.h

```diff
@@ -50,6 +50,79 @@ static inline void set_embedded_window_pointer(embed_window_pointer_t *wp, windo
 
 static inline void set_embedded_window_pointer(ignore_window_pointer_t *, window *) {}
 
+struct window
```

Comment:
I was going to do that on my first attempt but the number of files touched would have roughly doubled, since there are other forward declarations as struct (e.g. in mouse.h). I might do that in a later commit, since it also clarifies what dcx::window is being used as, or will be used as.